File: my-workout-api/src/main/java/com/avernet/myworkoutapi/history/History.java
package com.avernet.myworkoutapi.history;

import com.avernet.myworkoutapi.exercise.Exercise;
import com.avernet.myworkoutapi.workout.Workout;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class History {
    Long id;
    Float weight;
    Short reps;
    Boolean unilateral;
    Workout workout;
    Exercise exercise;
}

File: my-workout-app/src/interfaces/History.ts
import { Exercise } from './Exercise';
import { User } from './User';
import { Workout } from './Workout';

export interface History {
    id?: number;
    workout?: Workout;
    user?: User;
    exercise?: Exercise;
    weight?: number;
    reps?: number;
    unilateral?: boolean;
    duration?: number;
    restTime?: string;
}
